### PR TITLE
Add product variations list to new product management experience

### LINKED ISSUE #35772: Render product variations list
The variations for a product should be displayed beneath the options.

Note that some prior art exists for this with the attributes field, but we should evaluate whether or not it's worth it to separate and reuse that component or if we should just recreate a component specific to variations since the UI differs slightly.

https://github.com/woocommerce/woocommerce/blob/185d3337cac78ba12a4dd6c0384a82df32629cb2/plugins/woocommerce-admin/client/products/fields/attribute-field/attribute-field.tsx#L234-L289

Design:  BaoQ3PGPnhAiJCsJtxgej3-fi-103%3A42557&t=Jsh233onKmJpJYtN-0

Image

* [ ] A new form section for variations should be added with the appropriate text from the design
* [ ] All the product variations should appear in a sortable list
* [ ] Variation chips, price, and quantity should be shown in each item.  Other information such as the checkbox, visibility, and edit buttons will added in a separate issue
* [ ] A spinner should be shown when variations are loading

### PR DESCRIPTION
**Pull request by joshuatf**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds the variations list to the new product experience.

Note that this PR does not address persistence of ordering for variations or any other updates.  Those will be handled in a follow-up.

Screen Shot 2022-12-07 at 11 50 36 AM

Closes #35772 .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Add a product with variations using the old product experience
2. Navigate to that product in the new experience.  E.g., `wp-admin/admin.php?page=wc-admin&path=/product/{product_id}`
3. Click on "Options"
4. Note that the variations section exists with the correct text and a working link
5. Note that variations show a spinner until variations are loaded
6. Note that variations appear with their attributes, price, and stock status listed

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/products/fields/variations/variations.scss [hunk 1]
```diff
@@ -0,0 +1,39 @@
+.woocommerce-product-variations {
+    min-height: 300px;
```
Comment (joshuatf) [inline plugins/woocommerce-admin/client/products/fields/variations/variations.scss]: Note that this is temporary and height will be adjusted further in https://github.com/woocommerce/woocommerce/issues/35789.

File: plugins/woocommerce-admin/client/products/utils/get-product-stock-status.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/**
```
Comment (joshuatf) [inline plugins/woocommerce-admin/client/products/utils/get-product-stock-status.ts:1]: Hopefully this util isn't pushing the scope of this PR too far.  Happy to separate it into another PR if it helps.

File: plugins/woocommerce-admin/client/products/fields/variations/variations.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+					product_id: productId,
+				},
+			] ),
+			variations: getProductVariations< ProductVariation[] >( {
+				product_id: productId,
+			} ),
+		};
+	} );
+
+	if ( ! variations || isLoading ) {
```
Comment (joshuatf) [inline plugins/woocommerce-admin/client/products/fields/variations/variations.tsx:42]: Note that the current behavior is odd because some products don't have variations.  [Automatic generation of variations](https://github.com/woocommerce/woocommerce/issues/35778) will address this and hopefully guarantee that a variation always exists if any options exist.

File: plugins/woocommerce-admin/client/products/utils/get-product-stock-status.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+		'Out of stock',
+		'woocommerce'
+	),
+};
+
+/**
+ * Get the product stock quantity or stock status label.
+ *
+ * @param  product Product instance.
+ * @return {PRODUCT_STOCK_STATUS_KEYS} Product staus key.
```
Comment (octaedro) [inline plugins/woocommerce-admin/client/products/utils/get-product-stock-status.ts]: Two nitpick things:
- There is a typo in `staus`.
- Since we can also return a number, I think that this line should be like this:
```
@return {PRODUCT_STOCK_STATUS_KEYS|number} Product stock quantity or product status key.
```

**Review by octaedro — COMMENTED** [review]
Good job @joshuatf! This is testing well and the code looks good.
I just left a small comment, and it seems like there are a few small [lint errors](https://github.com/woocommerce/woocommerce/actions/runs/3642348200/jobs/6235367416) as well.

**Review by octaedro — APPROVED** [review]
Nice job @joshuatf! LGTM :shipit:
